server/metadata/cluster.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/tigrisdata/tigris/errors"
+	"github.com/tigrisdata/tigris/keys"
+	"github.com/tigrisdata/tigris/server/transaction"
+	ulog "github.com/tigrisdata/tigris/util/log"
+)
+
+// ClusterSubspace is used to store metadata about Tigris clusters.
+type ClusterSubspace struct {
+	metadataSubspace
+}
+
+var (
+	clusterVersion     = []byte{0x01}
+	clusterID          = uuid.Nil.String()
+	clusterMetadataKey = "key"
+)
+
+// ClusterMetadata keeps cluster wide metadata.
+type ClusterMetadata struct {
+	WorkerKeepalive time.Time
```

Comment:
is this for async?

Reply:
right. also just placeholder for now.

Reply:
Okay, because I think we need something on the collection level, also I was thinking if we should have async metadata subspace? But we can figure it out as we go.